pkg/southbound/synchronizer/factory.go

```diff
@@ -54,10 +56,12 @@ func Factory(topoChannel <-chan *topodevice.ListResponse, opStateChan chan<- eve
 			} else {
 				mStateGetMode = modelregistry.GetStateMode(mPlugin.GetStateMode())
 			}
+			operationalStateCacheLock.Lock()
 			operationalStateCache[notifiedDevice.ID] = make(devicechange.TypedValueMap)
 			target := southbound.NewTarget()
 			sync, err := New(ctx, notifiedDevice, opStateChan, southboundErrorChan,
 				operationalStateCache[notifiedDevice.ID], mReadOnlyPaths, target, mStateGetMode)
+			operationalStateCacheLock.Unlock()
```

Comment:
I think here you are keeping the lock for too long. By keeping it until the return of the New method you are not enabling the synchroniser to acquire the lock in order to populate the cache during startup and this might cause the failure you were seeing in the integration tests run. to me the Unlock shoudl happen after line 60, right after make.

Reply:
I think you should be passing the operationalStateCacheLock in tot he synchronizer with New() and replace the mutex created in there in synchronizer.go:65